SharpPcap/LibPcap/CaptureFileWriterDevice.cs

```diff
@@ -117,7 +117,15 @@ public override void Open(DeviceConfiguration configuration)
             }
 
             // set the device handle
-            PcapHandle = LibPcapSafeNativeMethods.pcap_open_dead((int)configuration.LinkLayerType, configuration.Snaplen);
+            if (configuration.TimestampResolution.HasValue)
+            {
+                PcapHandle = LibPcapSafeNativeMethods.pcap_open_dead_with_tstamp_precision((int)configuration.LinkLayerType,
```

Comment:
No check for `Pcap.LibpcapVersion >= new Version(1, 5, 1)` ?

Reply:
Yeah we should probably check that and report it, let me fix.